airflow/providers/http/sensors/http.py

```diff
@@ -90,7 +96,8 @@ def __init__(
         self.headers = headers or {}
         self.extra_options = extra_options or {}
         self.response_check = response_check
-
+        self.acceptable_status_code = acceptable_status_code or []
+        self.acceptable_status_code.append(404)
```

Comment:
This will add a new item every time the operator is initialized. When this operator is initialized multiple times, this element will also be added multiple times.
```
MY_API_ERROR_CODES=[500]
HttpSensor(acceptable_status_code=MY_API_ERROR_CODES)
HttpSensor(acceptable_status_code=MY_API_ERROR_CODES)

assert MY_API_ERROR_CODES = [500, 404, 404]
```

Reply:
Besides, I think you can achieve the desired effect if you pass your logic as parameter ``response_check``.  Have you tried this?

Reply:
I agree `response_check` is there for exactly that reason to allow multiple use-cases

Reply:
Yep, it's true. I take a deep look and notice the same now. Sorry

But in order for the `response_check` works for other codes than 404, it's necessary to fill the param `extra_option` with the value `{"check_response": False}`.

There is a reference for it in the hook method `run`
>         :param extra_options: additional options to be used when executing the request
>            i.e. {'check_response': False} to avoid checking raising exceptions on non
>            2XX or 3XX status codes

So, my changes are not necessary to reach this goal.

I will rewrite the tests for cases with `extra_option` and try to improve the doc section with this information of `{"check_response": False}` since it's an internal optional behavior in the hook.